spec/formatting.md

```diff
@@ -222,6 +222,61 @@ the following steps are taken:
 
    The form that resolved _operand_ and _option_ values take is implementation-defined.
 
+   A _local-declaration_ binds the resolved value of an _expression_
+   to a _variable_.
+   Thus, the output of one _function_ is potentially the _operand_
+   of another _function_. In other words, formatting functions
+   compose with each other.
+   For example, in
+   ```
+   .input {$n :number minIntegerDigits=3}
+   .local $n1 = {$n :number maxFractionDigits=3}
+   ```
+   the second call to `:number` composes with the first call.
+
+   In addition, selector functions compose with formatting functions
+   in the sense that a selector function's _operand_
+   may be the output of any formatting function.
+
+   Implementations SHOULD provide a means for formatting functions
+   to compose with each other
+   and for formatting functions to compose with selector functions.
+   Implementations that provide a means for defining custom functions
```

Comment:
I agree with the sense of this. We should also note somewhere that any two functions are not required to meaningfully compose; there is no requirement or expectation that the following make a meaningful composition:
```
.input {$date :datetime}
.local $person = {$date :x:personname}
```

Reply:
+1 to this. I'm not sure if I was able to explain this in yesterday's call, but I definitely agree that there should be no requirement for *any two* functions to meaningfully compose, as you've nicely put it. This is why I've been using the term *cooperative composition* (although I'm happy to call it something else), i.e. the functions must be aware of each other's interface in order to meaningfully compose.

Reply:
+100 although there are some different things going on here. In @macchiati's example, the types don't match. The `.local` declaration should emit an `Invalid Expression` error because `$date` isn't of a supported type for `:x:personname` (presumably).

In other cases, the types can match but the annotations might not be supported:

```
.input {$date :date style=long}
.local $foo = {$date :time}  // not style=long because style means "dateStyle"
```

For now, we should basically say something like "functions can decide what operands to support" and "functions can decide what functions or function options to support". We permit "composition" without requiring it **_or_** prohibiting it.

In the default registry we can add (in the TP period) some guidance for date/time/datetime and number functions as a guide.

Reply:
Agreed.